react-app/src/api/index.js

```diff
@@ -0,0 +1,86 @@
+import Account from 'account';
+import { createEntry } from 'utils/entry';
+
+const _HOST = 'http://localhost/';
+
+class ErrorCannotGetEntries extends Error {}
+
+class Api {
+  initAccount(username, keyPair) {
+    this._account = new Account(username, keyPair);
+  }
+
+  _getUrl(path) {
+    let url = `${_HOST}${path}`;
+    if (this._token) url += `?token=${this._token}`;
+
+    return url;
+  }
+
+  async _get(path) {
+    const url = this._getUrl(path);
+
+    return await fetch(url);
+  }
+
+  async _post(path, body) {
+    const url = this._getUrl(path);
+
+    const options = {
+      method: 'POST',
+      body
+    }
+
+    return await fetch(url, options);
+  }
+
+  async signUp() {
+    const data = this._account.toString();
+    const response = await this._post('signup', data);
+
+    return response.ok;
+  }
+
+  async signIn() {
+    const data = this._account.toString();
+    const response = await this._post('signin', data);
+
+    if (!response.ok) return false;
+
+    const receivedJson = await response.json();
+    const encryptedToken = receivedJson.EncryptedToken;
+    this._token = this._account.decrypt(encryptedToken);
+
+    return true;
+  }
+
+  async addEntry(entry) {
+    const cipherText = this._account.encrypt(entry.text);
+    const data = JSON.stringify({
+      Date: entry.date,
+      Ct: cipherText,
+    });
+
+    const response = await this._post('things', data);
+
+    return response.ok;
+  }
+
+  _decryptReceivedEntry = entry => createEntry(entry.Date, this._account.decrypt(entry.Content));
```

Comment:
So, the the `Date` is not encrypted?

Reply:
As far as I understood, it is not encrypted. In fact, I think that this can't be encrypted, if not, the server can't enumerate the entries filtered by a specific from-to period, am I right? :/

Reply:
Yes, that sounds about right.
@chiiph any insight to share here?

Reply:
Not sure what insight you would like. The date is the key to the item, so that's not secret. The only thing that is secret to the server is the contents of the items.